lib/logstash/filters/elasticsearch.rb

```diff
@@ -153,17 +170,23 @@ def self.validate_value(value, validator)
     return super(value, :uri)
   end
 
+  attr_reader :query_dsl
+
   def register
-    #Load query if it exists
-    if @query_template
-      if File.zero?(@query_template)
-        raise "template is empty"
-      end
-      file = File.open(@query_template, 'r')
-      @query_dsl = file.read
+    query_type = resolve_query_type
+    case query_type
+    when "esql"
+      invalid_params_with_esql = original_params.keys & %w(index query_template sort docinfo_fields aggregation_fields enable_sort result_size)
+      raise LogStash::ConfigurationError, "Configured #{invalid_params_with_esql} params cannot be used with ES|QL query" if invalid_params_with_esql.any?
```

Comment:
review note: raised a question if this can be ignored (with warn/info message) in es-input but I would go with strict to set _only_ required params for ES|QL.